### PR TITLE
Update discord-game-sdk to 3.2.1

### PR DESCRIPTION
**Pull request by L3nn0x**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by L3nn0x** [comment]
Hello, this is my first PR on this project, I would appreciate any comments you may have. Thank you very much for your time!

**Comment by L3nn0x** [comment]
@microsoft-github-policy-service agree

File: ports/discord-game-sdk/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "discord-game-sdk",
-  "version": "2.5.6",
-  "port-version": 5,
+  "version": "3.2.1",
   "description": "The Discord GameSDK is an easy drop-in SDK to help you manage all the hard things that come with making a game.",
   "homepage": "https://discord.com/developers/docs/game-sdk/sdk-starter-guide",
-  "supports": "((x64 & (windows | osx | linux)) | (x86 & windows)) & !uwp & !static",
+  "supports": "((x64 & (windows | osx | linux)) | (x86 & windows) | (arm64 & osx)) & !uwp & !static",
```
Comment (jimwang118) [inline ports/discord-game-sdk/vcpkg.json:6]: The upstream sees that the arm architecture support has been added. Does it only support `arm64-osx` here? Or is there corresponding support for other operating systems?

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

File: ports/discord-game-sdk/vcpkg.json (hunk 1 above)
Comment (rminderhoud) [reply ports/discord-game-sdk/vcpkg.json:6]: It appears Discord only publishes `.bundle` and `.dylib` for `aarch64` at this time

image

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
```

**Pull request merged**